pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -276,10 +320,37 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 		// only select virtualServices that matches a service
 		virtualServices = selectVirtualServices(virtualServices, servicesByName)
 	}
+
+	var routeCache *istio_route.Cache
+
+	if listenerPort > 0 {
+		services = make([]*model.Service, 0, len(servicesByName))
+		// sort services
+		for _, svc := range servicesByName {
+			services = append(services, svc)
+		}
+		sort.SliceStable(services, func(i, j int) bool {
+			return services[i].Hostname <= services[j].Hostname
+		})
+
+		routeCache = &istio_route.Cache{
+			RouteName:       routeName,
+			ListenerPort:    listenerPort,
+			Services:        services,
```

Comment:
I am somewhat concerned about the size of this key... There may be many services. I guess we wil have to check benchmarks

Reply:
Yes, we should use md5 like  ConfigKey.